src/normalize.js

```diff
@@ -92,21 +94,65 @@ function extractAllLinks(json, uri, opts) {
   return ret;
 }
 
+function extractToVirtualKey(uri, rel, content, contentKey) {
+  const ret = {};
+  const virtualKey = `${uri}#${rel}`;
+  ret[virtualKey] = {
+    [contentKey]: content,
+  };
+
+  ret[uri] = {};
+  ret[uri][rel] = {
+    href: virtualKey,
+    virtual: true,
+  };
+  return ret;
+}
+
 function mergeEmbeddedStandaloneCollections(embedded, links, opts) {
   const ret = {};
   merge(ret, links);
   merge(ret, embedded);
 
   keys(embedded).forEach((uri) => {
+    // check all embedded properties for embedded collections
     keys(embedded[uri]).forEach((rel) => {
-      if (Array.isArray(embedded[uri][rel]) && uri in links && rel in links[uri]) {
-        ret[uri][rel] = links[uri][rel];
-        ret[links[uri][rel].href] = {
-          [opts.embeddedStandaloneListKey]: embedded[uri][rel],
-          [opts.metaKey]: { self: links[uri][rel].href },
-        };
+      if (Array.isArray(embedded[uri][rel])) {
+        // standalone link provided (store embedded list as standalone link)
+        if (uri in links && rel in links[uri] && isSingleLink(links[uri][rel])) {
+          ret[uri][rel] = links[uri][rel];
+          ret[links[uri][rel].href] = {
+            [opts.embeddedStandaloneListKey]: embedded[uri][rel],
+            [opts.metaKey]: { self: links[uri][rel].href },
+          };
+        } else if (opts.embeddedStandaloneListVirtualKeys
+          && rel !== opts.embeddedStandaloneListKey) {
+          // no standalone link provided --> generate virtual key
+          delete ret[uri][rel];
+          merge(
+            ret,
+            extractToVirtualKey(uri, rel, embedded[uri][rel], opts.embeddedStandaloneListKey),
+          );
+        }
       }
     });
+
+    // also check remaining link properties to search for a possible collection
+    // which is not embedded
+    if (opts.embeddedStandaloneListVirtualKeys) {
+      difference(
+        keys(links[uri]),
+        [...keys(embedded[uri]), opts.embeddedStandaloneListKey],
+      ).forEach((rel) => {
+        if (Array.isArray(links[uri][rel])) {
```

Comment:
This if statement is missing test coverage for both the true and false case.

Reply:
Uh, I thought I had a test for this, but seems I added that one only in `hal-json-vuex`. Will add one.